components/ui/vertical-menu/component.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+import Button from 'components/ui/button';
+import Icon from 'components/ui/icon';
+
+import arrowIcon from 'assets/icons/arrow-down.svg?sprite';
+
+import './styles.scss';
+
+class VerticalMenu extends PureComponent {
+  render() {
+    const {
+      menu,
+      value: currentValue,
+      onClick,
+      className = '',
+      ...props
+    } = this.props;
+
+    return (
+      <ul
+        className={`c-vertical-menu ${className}`}
+        {...props}
+      >
+        {menu.map(({ label, value }) => (
+          <li className="menu-item">
+            <Button
+              theme=""
+              active={value === currentValue}
+              onClick={() => onClick(value)}
+            >
+              {label}
+              <Icon icon={arrowIcon} className="arrow-icon" />
+            </Button>
+          </li>
+        ))}
+      </ul>
+    )
+  }
+}
+
+VerticalMenu.propTypes = {
```

Comment:
As we are using PureComponent this could be called as a `static` on the class to be more efficient.

Reply:
good catch, i moved this in another component aswell `MapMenuSearch`

Reply:
Thanks!